### PR TITLE
MGMT-1059 Post register host transition keeps host role in case host …

### PR DESCRIPTION
**Pull request by YuviGold**
…exists

### CODE REVIEW

File: internal/host/transition.go [hunk 1]
```diff
@@ -44,12 +44,15 @@ func (th *transitionHandler) PostRegisterHost(sw stateswitch.StateSwitch, args s
...
+	// If host already exists
 	if err := th.db.First(&host, "id = ? and cluster_id = ?", sHost.host.ID, sHost.host.ClusterID).Error; err == nil {
 		currentState := swag.StringValue(host.Status)
 		host.Status = sHost.host.Status
+
+		// The reason for the double register is unknown (HW might have changed) -
+		// so we reset the hw info and start the discovery process again.
 		return updateHostStateWithParams(log, currentState, statusInfoDiscovering, &host, th.db,
-			"hardware_info", "", "role", "", "discovery_agent_version", params.discoveryAgentVersion)
+			"hardware_info", "", "role", sHost.host.Role, "discovery_agent_version", params.discoveryAgentVersion)
```
Comment (avishayt) [inline internal/host/transition.go:55]:
```suggestion
			"hardware_info", "", "discovery_agent_version", params.discoveryAgentVersion)
```

Comment (avishayt) [reply internal/host/transition.go:55]: Better to only set what is necessary to prevent potential races with other update API calls

Comment (filanov) [reply internal/host/transition.go:55]: +1

Comment (YuviGold) [reply internal/host/transition.go]: Fixed

File: internal/host/transition.go [hunk 2]
```diff
@@ -44,12 +44,21 @@ func (th *transitionHandler) PostRegisterHost(sw stateswitch.StateSwitch, args s
...
 	if err := th.db.First(&host, "id = ? and cluster_id = ?", sHost.host.ID, sHost.host.ClusterID).Error; err == nil {
 		currentState := swag.StringValue(host.Status)
 		host.Status = sHost.host.Status
-		return updateHostStateWithParams(log, currentState, statusInfoDiscovering, &host, th.db,
-			"hardware_info", "", "role", "", "discovery_agent_version", params.discoveryAgentVersion)
+
+		var updates []interface{}
+		updates = append(updates, "hardware_info", "", "discovery_agent_version", params.discoveryAgentVersion)
+
+		if sHost.host.Role != "" {
```
Comment (filanov) [inline internal/host/transition.go:55]: how will you get to this state? register API don't get role parameter. 
Suggesting just to delete this part.

Comment (YuviGold) [reply internal/host/transition.go:55]: Well it does seem simpler.

**Pull request merged**